src/bika/lims/api/__init__.py

```diff
@@ -1415,7 +1415,7 @@ def get_user_contact(user, contact_types=['Contact', 'LabContact']):
         return None
 
     from senaite.core.catalog import CONTACT_CATALOG  # Avoid circular import
-    query = {"portal_type": contact_types, "getUsername": user.id}
+    query = {"portal_type": contact_types, "getUsername": user.getId()}
```

Comment:
Yes I think so. Otherwise we have to get always the user(s) first, before we can say if one is linked or not. It could make sense as a separate column in the User&Groups Panel for instance.